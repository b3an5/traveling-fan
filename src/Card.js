import React from 'react';
import './css/Card.scss';
// import xIcon from './images/xIcon.svg'
import xIcon from './images/x-icon.png'

function Card(props) {

  return (
    <div className='card'>
      <img className='x-icon'
        onClick={props.showAllTeams}
        src={xIcon}
        alt="icon" />
      <img className="card-img"
        src={props.currentTeam.team_img}
        alt="NFL team" />
      <h2 className='team-info'>
        {props.currentTeam.name}</h2>
      <h2 className='team-info'>
        {props.currentTeam.city}, {props.currentTeam.state}</h2>
      <h2 className='team-info'>
        {props.currentTeam.division}</h2>
      <button className="card-btn"
        onClick={props.switchPopupView}>
        Show City Info</button>
      <h1 className='left-carot carot'
        onClick={props.rotateCarousel}>&laquo;</h1>
      <h1 className='right-carot carot'
        onClick={props.rotateCarousel}>&raquo;</h1>
    </div>
  )
}

export default Card;