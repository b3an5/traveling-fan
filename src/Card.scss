.card {
  border-radius: 5px;
  height: 95vh;
  width: 55%;
  text-align: center;
  background-color: white;
  border: solid black 5px;
  box-shadow: 2px 2px 8px 1px grey;
  margin: 0 auto;
  position: absolute;
  top: 20px;
}

.card-img {
  // height: 62%;
  width: 100%
}

.team-info {
  padding: 10px;
}

.x-icon {
  height: 20px;
  position: absolute;
  top: 0;
  left: 0;
}