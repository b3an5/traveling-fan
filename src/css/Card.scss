.card {
  border-radius: 5px;
  width: 45vw;
  text-align: center;
  background-color: white;
  border: solid black 5px;
  box-shadow: 2px 2px 8px 1px grey;
  margin: 0 auto;
  position: absolute;
  top: 185px;
  left: 27%;
  z-index: 3;
}

.card-img {
  width: 45vw;
}

.team-info {
  padding: 10px;
}


.x-icon {
    height: 60px;
    position: absolute;
    top: 0;
    left: 0;
    background-color: #F16857;
    border: 5px solid #F16857;
    border-radius: 60px;
    &:hover {
      // opacity: .7;
      background-color: white;
      border: 5px solid white;
    }
}

.x-icon:hover {
  cursor: pointer;
}

.carot {
  cursor: pointer;
  position: absolute;
  font-size: 3em;
  margin: 5px;
  padding: 85px 5px;
  &:hover {
    color: red;
  }
}

.left-carot {
  left: 0;
  top: 67%;
}

.right-carot {
  right: 0;
  top: 67%;
} 

.card-btn {
  margin: 10px auto;
  padding: 10px;
}