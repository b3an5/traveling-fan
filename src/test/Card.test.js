import React from 'react';
import Card from '../Card.js';
import { shallow } from 'enzyme';

const fakeNflTeam = {
  "name": "Arizona Cardinals",
  "state": "Arizona",
  "city": "Glendale",
  "team_icon": "http://assets.stickpng.com/thumbs/580b585b2edbce24c47b2aff.png",
  "division": "NFC West",
  "team_img": "https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg"
}

const fakeCity = {"attractions": ["Wildlife World Zoo", "White Tank Mountain Regional Park", "Westgate Arena", "Thunderbird Conservation Park"],
"city_image": "http://res.cloudinary.com/simpleview/image/upload/v1529338630/clients/phoenix/58468198_2b66_4f1e_8687_e26d986f678a_10c75a12-fe18-44d2-ab79-ed92fdb8ff6f.jpg",
"name": "Glendale",
"population": 226721,
"state": "Arizona"};

const mockShowAllTeams = jest.fn()
const mockSwitchPopupView = jest.fn()
const mockRotateCarousel = jest.fn()

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card 
        showAllTeams={mockShowAllTeams}
        switchPopupView={mockSwitchPopupView}
        currentTeam={fakeNflTeam}
        currentCity={fakeCity}
        rotateCarousel={mockRotateCarousel}
        key={fakeNflTeam.name} />
      )
  })

  it('should match the snapshot with all data passed correctly to the card', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click when the SHOW CITY INFO button is pressed', () => {
    wrapper.find('.card-btn').simulate('click')
    expect(mockSwitchPopupView).toBeCalled();
  })

  it('should register a click when the LEFT CARET is clicked', () => {
    wrapper.find('.left-caret').simulate('click')
    expect(mockRotateCarousel).toBeCalled()
  })

  it('should register a click when the RIGHT CARET is clicked', () => {
    wrapper.find('.right-caret').simulate('click')
    expect(mockRotateCarousel).toBeCalled()
  })
 
})