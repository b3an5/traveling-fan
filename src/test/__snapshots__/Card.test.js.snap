// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed correctly to the card 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cities={
      Array [
        Object {
          "attractions": Array [
            "Wildlife World Zoo",
            "White Tank Mountain Regional Park",
            "Westgate Arena",
            "Thunderbird Conservation Park",
          ],
          "city_image": "http://res.cloudinary.com/simpleview/image/upload/v1529338630/clients/phoenix/58468198_2b66_4f1e_8687_e26d986f678a_10c75a12-fe18-44d2-ab79-ed92fdb8ff6f.jpg",
          "name": "Glendale",
          "population": 226721,
          "state": "Arizona",
        },
      ]
    }
    clickedIndex={0}
    nflTeams={
      Array [
        Object {
          "city": "Glendale",
          "division": "NFC West",
          "name": "Arizona Cardinals",
          "state": "Arizona",
          "team_icon": "http://assets.stickpng.com/thumbs/580b585b2edbce24c47b2aff.png",
          "team_img": "https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg",
        },
      ]
    }
    toggleCardView={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="icon"
          className="x-icon"
          onClick={[MockFunction]}
          src="xIcon.svg"
        />,
        <img
          alt="NFL team"
          className="card-img"
          src="https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg"
        />,
        <h2
          className="team-info"
        >
          Arizona Cardinals
        </h2>,
        <h2
          className="team-info"
        >
          Glendale
          , 
          Arizona
        </h2>,
        <h2
          className="team-info"
        >
          NFC West
        </h2>,
        <button
          className="card-btn"
          onClick={[Function]}
        >
          Show City Info
        </button>,
        <h1
          className="left-carot carot"
        >
          «
        </h1>,
        <h1
          className="right-carot carot"
        >
          »
        </h1>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "icon",
          "className": "x-icon",
          "onClick": [MockFunction],
          "src": "xIcon.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "NFL team",
          "className": "card-img",
          "src": "https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Arizona Cardinals",
          "className": "team-info",
        },
        "ref": null,
        "rendered": "Arizona Cardinals",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Glendale",
            ", ",
            "Arizona",
          ],
          "className": "team-info",
        },
        "ref": null,
        "rendered": Array [
          "Glendale",
          ", ",
          "Arizona",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NFC West",
          "className": "team-info",
        },
        "ref": null,
        "rendered": "NFC West",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show City Info",
          "className": "card-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Show City Info",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "«",
          "className": "left-carot carot",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "«",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "»",
          "className": "right-carot carot",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "»",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="icon"
            className="x-icon"
            onClick={[MockFunction]}
            src="xIcon.svg"
          />,
          <img
            alt="NFL team"
            className="card-img"
            src="https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg"
          />,
          <h2
            className="team-info"
          >
            Arizona Cardinals
          </h2>,
          <h2
            className="team-info"
          >
            Glendale
            , 
            Arizona
          </h2>,
          <h2
            className="team-info"
          >
            NFC West
          </h2>,
          <button
            className="card-btn"
            onClick={[Function]}
          >
            Show City Info
          </button>,
          <h1
            className="left-carot carot"
          >
            «
          </h1>,
          <h1
            className="right-carot carot"
          >
            »
          </h1>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "icon",
            "className": "x-icon",
            "onClick": [MockFunction],
            "src": "xIcon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "NFL team",
            "className": "card-img",
            "src": "https://cdn.vox-cdn.com/thumbor/M5Gx-DCa9eRAqw_aqIyuG6I2wDo=/0x0:4777x3021/1200x800/filters:focal(2094x550:2858x1314)/cdn.vox-cdn.com/uploads/chorus_image/image/62274919/usa_today_11648254.0.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Arizona Cardinals",
            "className": "team-info",
          },
          "ref": null,
          "rendered": "Arizona Cardinals",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Glendale",
              ", ",
              "Arizona",
            ],
            "className": "team-info",
          },
          "ref": null,
          "rendered": Array [
            "Glendale",
            ", ",
            "Arizona",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NFC West",
            "className": "team-info",
          },
          "ref": null,
          "rendered": "NFC West",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show City Info",
            "className": "card-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Show City Info",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "«",
            "className": "left-carot carot",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "«",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "»",
            "className": "right-carot carot",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "»",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
